dramatiq/middleware/declare_queues.py

```diff
@@ -0,0 +1,9 @@
+from .middleware import Middleware
+
+
+class DeclareQueuesMiddleware(Middleware):
+    def before_consume(self, broker):
+        broker.declare_prepared_queues()
+
+    def before_enqueue(self, broker, *_):
+        broker.declare_prepared_queues()
```

Comment:
Will send N `declare_queues` to rabbitmq before every `.send()`.
Maybe you should check if the queues have been declared before.